### DISCUSSION TITLE
Types for $page.data

### DISCUSSION THREAD

**Discussion by jdgamble555** [sveltejs/kit#8219] [Q&A]
Is there a way to use the `PageData` types in the `$page.data`  variable?

I can do this in the route component:

```js
import type { LayoutData } from './$types';
export let data: LayoutData;
```

But how do I get matching types in a child component (specifically to work with $page.data)? 

There is no way to import the proper types. These do NOT work:

```js
import type { PageData } from '../../$types';
```
or 
```js
import type { PageData } from './$types';
```

It seems `$page.data.X` defaults to `any` type, which is not desirable.

J

**Comment by richarddavenport** [comment]
From the [docs](https://kit.svelte.dev/docs/types#app-pagedata):
> Defines the common shape of the [$page.data store](https://kit.svelte.dev/docs/modules#$app-stores-page) - that is, the data that is shared between all pages.

**Reply by jdgamble555** [reply]
Yeah, it should work the same as PageData though.
